collections/_api_standards/3-api-iterating.md

```diff
@@ -0,0 +1,60 @@
+---
+layout: default
+title:  "Iterating your API"
+category: api_standards
+---
+
+## Iterating your API
+
+In all but the simplest of domains, future changes to your API are inevitable.  Right from the point we deploy our new bright and shiny API, we know before too long we will need to change it.  We will then be faced with all the usual issues about how we manage that change without striking terror into the hearts of our API consumers.  In short, we need to think about how we  manage this change up-front, before we get into problems.  Some standards are provided below (these apply to both inner and outer APIs).
+
+### Don't break it in the first place
+
+Consider applying [Postel's Law](https://michaelfeathers.silvrback.com/the-universality-of-postel-s-law), and...
+
+> Be liberal in what you accept, and conservative in what you send.
+
+By adopting this philosophy (and encouraging consumers to adopt it), you will be more resilient to breaking changes in the first place.  
+
+**As the API provider**, think through your resource definitions.  As previously stated, use domain language and be consistent.  Whilst you want the API to be rich and fully featured, avoid the temptation to publish any resources (or properties of resources) pertaining to concepts which are not 100% concrete (which may later change / be removed).  As soon as you publish a resource definition clients will become dependant on it.  If a published resource contains data which is transient or not thought through, when you decide to remove / change it, it will make a breaking change.  So try and avoid doing it, as _Postel says, be conservative in what you send_.
+
+**As an API consumer**, think about what dependencies you take upon the API, and ensure your code is only dependant on those aspects of the API it actually cares about.  For example, if you consume an API which returns information on locations of courses, and all you need is the post code and course name, then only take a dependency on those items.  If anything gets added or removed outside of those two items, you should continue to work without issue.  Avoid solutions which serialise entire resources into strongly typed classes, when you do not care about the majority of the properties.  As Postel says, _be liberal in what you accept_.
+
+## Versioning your API

```

Comment:
I'd like us to define a few more 'guidelines' around good practice. 

When to increment a major version and when to increment the minor version.